great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -273,21 +274,15 @@ def save_expectation_suite(
         Returns:
             None
         """
-        ge_cloud_id = expectation_suite.ge_cloud_id
         key = GeCloudIdentifier(
             resource_type=GeCloudRESTResource.EXPECTATION_SUITE,
-            ge_cloud_id=ge_cloud_id
-            if ge_cloud_id is not None
-            else str(expectation_suite.ge_cloud_id),
+            ge_cloud_id=expectation_suite.ge_cloud_id,
+            resource_name=expectation_suite.expectation_suite_name,
         )
-        if (
-            self.expectations_store.has_key(key)  # noqa: W601
-            and not overwrite_existing
-        ):
-            raise ge_exceptions.DataContextError(
-                f"expectation_suite with GE Cloud ID {ge_cloud_id} already exists. "
-                f"If you would like to overwrite this expectation_suite, set overwrite_existing=True."
-            )
+
+        if not overwrite_existing:
+            self._validate_suite_unique_constaints_before_save(key)
```

Comment:
Since save works with both new suites and existing ones, we want to be able to ensure that we're respecting the unique constraint on both ids and names. If using a new suite, we want to check against the name and if using an existing one, we want to check against the ID.

Let me know if that makes sense!